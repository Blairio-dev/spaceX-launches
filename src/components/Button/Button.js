import React from "react";
import PropTypes from "prop-types";
import styled from "@emotion/styled";
import { colours } from "../../assets/tokens";
import { Image } from "../../components";
import RefreshIcon from "../../assets/icon/refresh.png";
import RefreshIcon2x from "../../assets/icon/refresh@2x.png";
import RefreshIcon3x from "../../assets/icon/refresh@3x.png";
import SortIcon from "../../assets/icon/sort.png";
import SortIcon2x from "../../assets/icon/sort.png";
import SortIcon3x from "../../assets/icon/sort@2x.png";
import SelectIcon from "../../assets/icon/select@3x.png";
import SelectIcon2x from "../../assets/icon/select@2x.png";
import SelectIcon3x from "../../assets/icon/select@3x.png";

const StyledButton = styled("button")`
  align-items: center;
  background: ${colours.primaryBlue};
  border: 2px solid ${colours.primaryBlue};
  ${(props) =>
    props.displayType === "roundedLeft" && "border-radius: 21.5px 0 0 21.5px;"}
  color: ${colours.white};
  cursor: pointer;
  display: inline-flex;
  height: 43px;
  margin: 0;
  padding: 0 16px;
  transition: all 0.05s ease-out;

  :focus,
  :hover {
    border-color: ${colours.grey};
    outline: 0;
  }

  span {
    font-family: "Brandon Grotesque Bold";
    font-size: 16px;
    margin-top: 0;
  }

  img {
    margin-left: 8px;
  }
`;

const basePropTypes = {
  onClick: PropTypes.func.isRequired,
};

const Base = ({
  displayType,
  iconHeight,
  iconSourceSet,
  labelText,
  onClick,
}) => (
  <StyledButton displayType={displayType} onClick={onClick} type="button">
    <span>{labelText}</span>
    <Image
      alt="Rocket launching"
      height={iconHeight}
      sourceSet={iconSourceSet}
    />
  </StyledButton>
);

Base.propTypes = {
  ...basePropTypes,
  displayType: PropTypes.oneOf(["standard", "roundedLeft"]).isRequired,
  iconHeight: PropTypes.string.isRequired,
  iconSourceSet: PropTypes.node.isRequired,
  labelText: PropTypes.string.isRequired,
};

const Filter = ({ labelText, onClick }) => (
  <Base
    displayType="standard"
    iconHeight="5px"
    iconSourceSet={`
      ${SelectIcon3x} 3x,
      ${SelectIcon2x} 2x,
      ${SelectIcon} 1x,
    `}
    labelText={labelText}
    onClick={onClick}
  />
);

Filter.propTypes = {
  ...basePropTypes,
  labelText: PropTypes.string.isRequired,
};

const Reload = ({ onClick }) => (
  <Base
    displayType="roundedLeft"
    iconHeight="14px"
    iconSourceSet={`
      ${RefreshIcon3x} 3x,
      ${RefreshIcon2x} 2x,
      ${RefreshIcon} 1x,
    `}
    labelText="Reload Data"
    onClick={onClick}
  />
);

Reload.propTypes = {
  ...basePropTypes,
};

const Sort = ({ labelText, onClick }) => (
  <Base
    displayType="standard"
    iconHeight="18px"
    iconSourceSet={`
      ${SortIcon3x} 3x,
      ${SortIcon2x} 2x,
      ${SortIcon} 1x,
    `}
    labelText={labelText}
    onClick={onClick}
  />
);

Sort.propTypes = {
  ...basePropTypes,
  labelText: PropTypes.string.isRequired,
};

const Button = {
  Filter,
  Reload,
  Sort,
};

export { Button };
