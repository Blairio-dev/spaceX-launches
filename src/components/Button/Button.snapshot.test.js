import React from "react";
import renderer from "react-test-renderer";
import { Button } from "./Button";

const DEFAULT_PROPS = {
  onClick: () => {},
};

describe("Button Snapshots", () => {
  describe(".Reload", () => {
    it("Renders with default props", () => {
      const tree = renderer
        .create(<Button.Reload {...DEFAULT_PROPS} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    describe(".Sort", () => {
      it("Renders with default props", () => {
        const tree = renderer
          .create(<Button.Sort {...DEFAULT_PROPS} labelText="Sort" />)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
