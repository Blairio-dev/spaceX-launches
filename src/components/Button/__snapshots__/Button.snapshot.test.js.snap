// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshots .Reload .Sort Renders with default props 1`] = `
<button
  className="css-4otj2c"
  onClick={[Function]}
  type="button"
>
  <span>
    Sort
  </span>
  <img
    alt="Rocket launching"
    height="18px"
    srcSet="
      sort@2x.png 3x,
      sort.png 2x,
      sort.png 1x,
    "
  />
</button>
`;

exports[`Button Snapshots .Reload Renders with default props 1`] = `
<button
  className="css-adslba"
  onClick={[Function]}
  type="button"
>
  <span>
    Reload Data
  </span>
  <img
    alt="Rocket launching"
    height="14px"
    srcSet="
      refresh@3x.png 3x,
      refresh@2x.png 2x,
      refresh.png 1x,
    "
  />
</button>
`;
